projects/samples/contests/robocup/controllers/player/player.cpp

```diff
@@ -392,51 +426,21 @@ class PlayerServer {
       webots::Device *device = robot->getDevice(sensorTimeStep.name());
       if (device) {
         const int sensor_time_step = sensorTimeStep.timestep();
-        if (sensor_time_step) {
-          if (sensors.count(device) == 0)
-            new_sensors.insert(device);
-        } else
-          sensors.erase(device);
 
-        if (sensor_time_step != 0 && sensor_time_step < basic_time_step)
+        if (sensor_time_step == 0)
+          sensors.erase(device);
+        else if (sensor_time_step != 0 && sensor_time_step < basic_time_step)
```

Comment:
With the way code is written, we can just remove the `sensor_time_step != 0 &&` right?

Reply:
Yes, totally. I fixed this!